# A0 — vectors and matrices

Rotate P=(2,1) by 45deg CCW around origin, then translate by (1,2).
Use **homogeneous coordinates** so rotation and translation compose into one matrix.

In [2]:
import numpy as np

theta = np.deg2rad(45)
c, s = np.cos(theta), np.sin(theta)

R = np.array([[c, -s, 0],
              [s,  c, 0],
              [0,  0, 1]])
T = np.array([[1, 0, 1],
              [0, 1, 2],
              [0, 0, 1]])

M = T @ R   # translate * rotate: rotation applied first
P = np.array([2.0, 1.0, 1.0])
Q = M @ P
print('Q =', Q[:2])


Q = [1.70710678 4.12132034]


Sanity check: rotate (2,1) by 45deg = (2cos45 - sin45, 2sin45 + cos45) = (0.707, 2.121). Add (1,2) -> (1.707, 4.121). Matches.